src/main/java/com/surrealdb/config/SurrealCrudRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.surrealdb.config;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface SurrealCrudRepository<T, ID> {
```

Comment:
This is a nice convenience, but it goes beyond what the driver is.

That said the driver in Rust does provide a DSL-like experience, so this fits under that.

In order not to pollute the driver codebase further, I would recommend having such patterns outside the driver in a separate repository. Otherwise this becomes bundled with the driver and we have the burden of maintaining it - providing variations and fitting it into all manner of APIs. That isn't the purpose of the driver.